lib/src/asset/asset.dart

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+library build.src.asset.asset;
+
+import 'id.dart';
+
+/// A fully realized asset whose content is available synchronously.
+class Asset {
+  /// The id for this asset.
+  final AssetId id;
+
+  /// The content for this asset.
+  final String contents;
```

Comment:
Definitely an open question, I don't know of any transformers today which read in by bytes. Primarily I made the decision to only support String apis to start, but I have been using `*AsString` in other api so I should probably at least change this to `stringContents` or `contentsAsString`.

Basically, I don't want to exclude the possibility of supporting `Stream<int>` later on but I also don't want to actually add support for it until somebody actually needs it.